decompiler/pipeline/dataflowanalysis/dead_path_elimination.py

```diff
@@ -80,6 +96,32 @@ def _is_invalid_edge(self, edge: BasicBlockEdge, condition: BoolRef) -> bool:
             condition = Not(condition)
         return self._logic_converter.is_not_satisfiable(condition, timeout=self._timeout)
 
+    def _is_invalid_edge_delogic(self, edge: BasicBlockEdge, condition: WorldObject) -> bool:
+        """
+        Check whether the condition of the given branch is satisfyable or not.
+
+        Returns a tuple of bools indicating the viability of the true and false branches.
+        """
+        if isinstance(edge, FalseCase):
+            condition = self._logic_converter.negate(condition)
+        result = self._logic_converter.make_variable(condition)
+        result.simplify()
+        # simplify to fixpoint.
+        while True:
+            if hasattr(result, "operands"):
+                result_copy = self._logic_converter.make_variable(result.copy_tree())
+                result.simplify()
+                if self._logic_converter.compare(result_copy, result):
+                    break
+            else:
+                break
```

Comment:
I am a bit afraid of endless loops and simplifications. I do not know whether there could be the case that we simplify between two equivalent expressions. 
```suggestion
        for _ in range(some_bound):
            if isinstance(result, Operation):
                result_copy = self._logic_converter.make_variable(result.copy_tree())
                result.simplify()
                if self._logic_converter.compare(result_copy, result):
                    break
            else:
                break
```

Reply:
Ah, I assumed that a fixpoint would always be reached :x Will use the ```timeout``` parameter to constraint the looping.